File: 0036-valid-sudoku/0036-valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        for(int i = 0 ; i < 9 ; i++)
        {
            unordered_set <char> s1;
            unordered_set <char> s2;
            for(int j =0 ; j < 9 ;j++)
            {
                if(board[i][j]!='.' && s1.count(board[i][j]))
                    return false;
                if(board[j][i]!='.' && s2.count(board[j][i]))
                    return false;
                s1.insert(board[i][j]);
                s2.insert(board[j][i]);        
            }
            s1.clear();
            s2.clear();
            
        }
        
        int i = 0,j = 0;
        while(i<9)
        {    
            unordered_set <char> s;
            for(int x = i ; x <i+3 ; x++)
            {            
                for(int y =j ; y < j+3 ;y++)
                {
                    if(board[x][y]>='1'&& board[x][y]<='9' && s.count(board[x][y]))
                        return false;
                    s.insert(board[x][y]);
                }
            }
            s.clear();
            j+=3;
            if(j>=9)
            {
                i+=3;
                j=0;
            }
            
            
            
        }
        
        return true;
        
    }
};

File: 0036-valid-sudoku/0036-valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {

        Boolean validCol = isValidCol(board);

        if (!validCol) return false;
        Boolean validRow = isValidRow(board);
        if (!validRow) return false;
        Boolean checkinnerSquares = checkinnerSquares(board);
        if (!checkinnerSquares) return false;

        return true;
    }


    boolean isDigit(char c) {
        return Character.isDigit(c);
    }

    private Boolean isValidCol(char[][] board) {
        for (int j = 0; j < 9; j++) {
            HashSet<Integer> hashSet = new HashSet<>();
            for (int i = 0; i < 9; i++) {
                char c = board[i][j];
                if (isDigit(c) && hashSet.contains(Integer.parseInt(String.valueOf(c)))) {
                    return false;
                } else if (isDigit(c)) {
                    hashSet.add(Integer.parseInt(String.valueOf(c)));
                }
            }


        }
        return true;

    }


    boolean isValidRow(char[][] board) {
        for (int j = 0; j < 9; j++) {
            HashSet<Integer> hashSet = new HashSet<>();
            for (int i = 0; i < 9; i++) {
                char c = board[j][i];
                if (isDigit(c) && hashSet.contains(Integer.parseInt(String.valueOf(c)))) {
                    return false;
                } else if (isDigit(c)) {
                    hashSet.add(Integer.parseInt(String.valueOf(c)));
                }
            }


        }
        return true;
    }

    boolean checkinnerSquares(char[][] board) {
        for (int boxes = 0; boxes < 9; boxes++) {
            HashSet<Integer> hashSet = new HashSet<>();
            for (int i = (boxes / 3) * 3; i <= (2 + (boxes / 3) * 3); i++) {
                for (int j = (boxes % 3) * 3; j <= 2 + ((boxes % 3) * 3); j++) {
                    char c = board[i][j];
                    if (isDigit(c) && hashSet.contains(Integer.parseInt(String.valueOf(c)))) {
                        return false;
                    } else if (isDigit(c)) {
                        hashSet.add(Integer.parseInt(String.valueOf(c)));
                    }
                }


            }


        }

        return true;
    }


}
